### PR TITLE
Drop CircleCI and switch to GitHub Actions

### PR DESCRIPTION
**Pull request by pi-314159**
The title says it all.

### CODE REVIEW

**Comment by pi-314159** [comment]
@baentsch GHCI passed! Could you remove circleci from this repo? I don't have access to settings of this repo.

**Comment by baentsch** [comment]
Thanks for this improvement!

> @baentsch GHCI passed! Could you remove circleci from this repo? I don't have access to settings of this repo.

I'd love (and even know how :) to but Sorry, this seems to be yet another area where LF removed permissions when they took ownership of OQS effectively neutering me/relegating me to be "maintainer in name only"; apparently another line item for https://github.com/open-quantum-safe/tsc/issues/45, @dstebila

image

Tagging @ryjones to consider clicking the "Stop Building" button above.

File: .github/workflows/shared.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+      - 'oqs_template/**'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - 'oqs_template/**'
+  schedule:
```
Comment (baentsch) [inline .github/workflows/shared.yml:14]: Same question as for demos: Why a scheduled build?

Comment (pi-314159) [reply .github/workflows/shared.yml:14]: to detect issues early, as updates to dependencies such as libunwind and Go can result in build failures.

File: .github/workflows/static.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - master
+    paths-ignore:
+      - '**.md'
+      - 'oqs_template/**'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - 'oqs_template/**'
+  schedule:
+    - cron: '2 7 18,28 * *'
```
Comment (baentsch) [inline .github/workflows/static.yml:15]: Why scheduled?

Comment (pi-314159) [reply .github/workflows/static.yml:15]: to detect issues early, as updates to dependencies such as libunwind and Go can result in build failures

File: oqs_scripts/try_handshake.py [hunk 3]
```diff
@@ -96,20 +95,19 @@
 def try_handshake(bssl):
     random_sig = random.choice(sigs)
     server = subprocess.Popen([bssl, 'server',
-                                     '-accept', '0',
+                                     '-accept', '31416',
```
Comment (baentsch) [inline oqs_scripts/try_handshake.py:98]: Isn't this risky? If run on a test machine that coincidentally has this port taken, CI will fail "surprisingly", no?

File: oqs_scripts/try_handshake.py [hunk 4]
```diff
@@ -96,20 +95,19 @@
...
     # The server should (hopefully?) start
     # in 10 seconds.
     time.sleep(10)
-    server_port = psutil.Process(server.pid).connections()[0].laddr.port
 
     # Try to connect to it with the client
     random_kex = random.choice(kexs)
     client = subprocess.run([bssl, 'client',
-                                   '-connect', 'localhost:{}'.format(str(server_port)),
+                                   '-connect', 'localhost:31416',
```
Comment (baentsch) [inline oqs_scripts/try_handshake.py:110]: see question above on specific port usage

File: oqs_scripts/try_handshake.py (hunk 3 above)
Comment (pi-314159) [reply oqs_scripts/try_handshake.py:98]: I don't believe any known services use this port: https://www.speedguide.net/port.php?port=31416.

Additionally, `ubuntu-latest` image does not have any services using this port, so it's safe.

File: .github/PULL_REQUEST_TEMPLATE [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
+Thank you for contributing to OQS-BoringSSL! Before submitting your pull request, please ensure the following:
 
-We do, however, take contributions gladly.
+1. Your changes include new tests or modifications to existing tests as needed.
 
-See https://boringssl.googlesource.com/boringssl/+/master/CONTRIBUTING.md
+2. You have tested the changes in a clean build environment.
 
-Thanks!
+3. Your changes do not break the build for projects that depend on OQS-BoringSSL.
```
Comment (baentsch) [inline .github/PULL_REQUEST_TEMPLATE:7]: I like that wording. For the "uninitiated", would it be sensible to create a pointer (at least) to oqs-demos (curl/nginx-quic), maybe even a script a new contributor could run so any PR doesn't break something there? Or even a downstream-test-build trigger for CI?

Comment (pi-314159) [reply .github/PULL_REQUEST_TEMPLATE]: updated

File: oqs_scripts/try_handshake.py (hunk 3 above)
Comment (baentsch) [reply oqs_scripts/try_handshake.py]: Thanks for the background check. It still "feels" unclean (I hate hard-coding stuff as that always carries the potential to "bite" me in the future). But a) you're using a mathematician's holy number, then of course, this is "Good" and b) it'll bite _you_ :-)

File: .github/PULL_REQUEST_TEMPLATE (hunk 5 above)
Comment (baentsch) [reply .github/PULL_REQUEST_TEMPLATE]: Please do with hyperlinks to oqs-demos. Just using the terms here may not be helpful for the "uninitiated" (not knowing about where/what's in oqs-demos). Apologies for being a pain-in-the-butt, but (one-time) clarity removes (many-time) pointless questions.

Comment (pi-314159) [reply .github/PULL_REQUEST_TEMPLATE]: @baentsch Done

**Review by baentsch — APPROVED** [review]
Thanks for this move off CCI!

**Pull request merged**